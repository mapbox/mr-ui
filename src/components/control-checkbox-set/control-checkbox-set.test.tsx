import React from 'react';
import { shallow } from 'enzyme';
import ControlCheckboxSet from './control-checkbox-set';

describe('ControlCheckbox', () => {

  /*
testCases.basic = {
  description: 'basic',
  component: ControlCheckboxSet,
  props: {
    id: 'testinput',
    options: [
      { label: 'Option 1', value: 'Option value 1' },
      { label: 'Option 2', value: 'Option value 2' }
    ],
    onChange: jest.fn()
  }
};

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('onChange gets called with input id and value', () => {
      const mockEventOne = {
        target: {
          value: 'foo'
        }
      };
      const mockEventTwo = {
        target: {
          value: 'foobar'
        }
      };
      wrapper.find('input').first().props().onChange(mockEventOne);
      wrapper.find('input').first().props().onChange(mockEventTwo);
      wrapper.find('input').first().props().onChange(mockEventOne);
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
      expect(testCase.props.onChange).toHaveBeenCalledTimes(3);
      expect(testCase.props.onChange).toHaveBeenCalledWith(
        ['foobar'],
        'testinput'
      );
    });
  });

testCases.disabled = {
  description: 'disabled',
  component: ControlCheckboxSet,
  props: {
    id: 'testinput',
    options: [
      { label: 'Option 1', value: 'Option value 1' },
      { label: 'Option 2', value: 'Option value 2', disabled: true }
    ],
    onChange: jest.fn()
  }
};

  describe(testCases.disabled.description, () => {
    beforeEach(() => {
      testCase = testCases.disabled;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('extra props works', () => {
      const props = wrapper.find('input').last().props();
      expect(props).toHaveProperty('disabled', true);
    });
  });

testCases.allOptions = {
  description: 'all options',
  component: ControlCheckboxSet,
  props: {
    id: 'testinput',
    options: [
      { label: <span>Option 1</span>, value: 'Option value 1' },
      { label: <span>Option 2</span>, value: 'Option value 2' },
      { label: <span>Option 3</span>, value: 'Option value 3' }
    ],
    validationError: 'oh no!',
    value: ['Option value 1', 'Option value 2'],
    autoFocus: true,
    optional: true,
    themeCheckboxContainer: 'w-full',
    themeCheckbox: 'checkbox--red mr6',
    themeControlWrapper: 'bg-red-light',
    onChange: jest.fn()
  }
};
  describe(testCases.allOptions.description, () => {
    beforeEach(() => {
      testCase = testCases.allOptions;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
*/
});
