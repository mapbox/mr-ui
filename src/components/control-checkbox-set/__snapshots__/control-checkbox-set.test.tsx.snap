// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlCheckbox all options renders 1`] = `
<body>
  <div>
    <div
      class="relative bg-red-light"
    >
      <div
        aria-invalid="true"
        data-testid="testinput-input"
        id="testinput"
      >
        <label
          class="checkbox-container w-full cursor-pointer"
        >
          <input
            aria-checked="true"
            checked=""
            data-testid="testinput-0-input"
            name="testinput"
            type="checkbox"
            value="Option value 1"
          />
          <div
            class="checkbox--red mr6 checkbox"
          >
            <svg
              aria-hidden="true"
              class="events-none icon"
              data-testid="icon-check"
              focusable="false"
              style="width: 18px; height: 18px;"
            >
              <use
                xlink:href="#icon-check"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              />
            </svg>
            <span
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            >
              check
            </span>
          </div>
          <span>
            Option 1
          </span>
        </label>
        <label
          class="checkbox-container w-full cursor-pointer"
        >
          <input
            aria-checked="true"
            checked=""
            data-testid="testinput-1-input"
            name="testinput"
            type="checkbox"
            value="Option value 2"
          />
          <div
            class="checkbox--red mr6 checkbox"
          >
            <svg
              aria-hidden="true"
              class="events-none icon"
              data-testid="icon-check"
              focusable="false"
              style="width: 18px; height: 18px;"
            >
              <use
                xlink:href="#icon-check"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              />
            </svg>
            <span
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            >
              check
            </span>
          </div>
          <span>
            Option 2
          </span>
        </label>
        <label
          class="checkbox-container w-full cursor-pointer"
        >
          <input
            aria-checked="false"
            data-testid="testinput-2-input"
            name="testinput"
            type="checkbox"
            value="Option value 3"
          />
          <div
            class="checkbox--red mr6 checkbox"
          >
            <svg
              aria-hidden="true"
              class="events-none icon"
              data-testid="icon-check"
              focusable="false"
              style="width: 18px; height: 18px;"
            >
              <use
                xlink:href="#icon-check"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              />
            </svg>
            <span
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            >
              check
            </span>
          </div>
          <span>
            Option 3
          </span>
        </label>
      </div>
      <div
        role="alert"
      >
        <div
          class="bg txt-s mt3 py6 px12 bg-red-faint round border border--red color-red"
          data-testid="testinput-error"
          id="testinput-error"
        >
          oh no!
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ControlCheckbox basic renders 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <div
        data-testid="testinput-input"
        id="testinput"
      >
        <label
          class="checkbox-container txt-s block mb6 flex cursor-pointer"
        >
          <input
            aria-checked="false"
            data-testid="testinput-0-input"
            name="testinput"
            type="checkbox"
            value="Option value 1"
          />
          <div
            class="mr6 inline-block checkbox--blue checkbox--s-label checkbox"
          >
            <svg
              aria-hidden="true"
              class="events-none icon"
              data-testid="icon-check"
              focusable="false"
              style="width: 18px; height: 18px;"
            >
              <use
                xlink:href="#icon-check"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              />
            </svg>
            <span
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            >
              check
            </span>
          </div>
          Option 1
        </label>
        <label
          class="checkbox-container txt-s block mb6 flex cursor-pointer"
        >
          <input
            aria-checked="false"
            data-testid="testinput-1-input"
            name="testinput"
            type="checkbox"
            value="Option value 2"
          />
          <div
            class="mr6 inline-block checkbox--blue checkbox--s-label checkbox"
          >
            <svg
              aria-hidden="true"
              class="events-none icon"
              data-testid="icon-check"
              focusable="false"
              style="width: 18px; height: 18px;"
            >
              <use
                xlink:href="#icon-check"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              />
            </svg>
            <span
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            >
              check
            </span>
          </div>
          Option 2
        </label>
      </div>
      <div
        role="alert"
      />
    </div>
  </div>
</body>
`;

exports[`ControlCheckbox disabled renders 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <div
        data-testid="testinput-input"
        id="testinput"
      >
        <label
          class="checkbox-container txt-s block mb6 flex"
        >
          <input
            aria-checked="false"
            data-testid="testinput-0-input"
            disabled=""
            name="testinput"
            type="checkbox"
            value="Option value 1"
          />
          <div
            class="mr6 inline-block checkbox--blue checkbox--s-label checkbox"
          >
            <svg
              aria-hidden="true"
              class="events-none icon"
              data-testid="icon-check"
              focusable="false"
              style="width: 18px; height: 18px;"
            >
              <use
                xlink:href="#icon-check"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              />
            </svg>
            <span
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            >
              check
            </span>
          </div>
          Option 1
        </label>
        <label
          class="checkbox-container txt-s block mb6 flex cursor-pointer"
        >
          <input
            aria-checked="false"
            data-testid="testinput-1-input"
            name="testinput"
            type="checkbox"
            value="Option value 2"
          />
          <div
            class="mr6 inline-block checkbox--blue checkbox--s-label checkbox"
          >
            <svg
              aria-hidden="true"
              class="events-none icon"
              data-testid="icon-check"
              focusable="false"
              style="width: 18px; height: 18px;"
            >
              <use
                xlink:href="#icon-check"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              />
            </svg>
            <span
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            >
              check
            </span>
          </div>
          Option 2
        </label>
      </div>
      <div
        role="alert"
      />
    </div>
  </div>
</body>
`;
